src/odemis/acq/fastem.py

```diff
@@ -452,6 +397,77 @@ def acquire_roa(self, dataflow):
+            raise ModuleNotFoundError("Need fastem_calibrations repository to run pre-calibrations.")
+
+        try:
+            fastem_conf.configure_scanner(self._scanner, fastem_conf.MEGAFIELD_MODE)
+
+            # Set the beamshift to zero before adjusting the descanner offset, this ensures that the maximum beamshift
+            # range can be utilized when the pattern is centered on the mppc detector.
+            self._beamshift.shift.value = (0, 0)
+
+            logging.debug("Read initial Hw settings.")
+            asm_config_orig = configure_hw.get_config_asm(self._multibeam, self._descanner, self._detector)
+            # Autofocus multiprobe ensures the beams are focused at the start of the ROA acquisition.
+            logging.debug("Running optical autofocus before ROA acquisition.")
+            autofocus_multiprobe.run_autofocus(self._scanner,
+                                               self._multibeam,
+                                               self._descanner,
+                                               self._detector,
+                                               self._detector.data,
+                                               self._ccd,
+                                               self._stage)
+
+            configure_hw.configure_asm(self._multibeam,
+                                       self._descanner,
+                                       self._detector,
+                                       self._detector.data,
+                                       asm_config_orig,
+                                       upload=False)
+            # Image translation pre-alignment ensures that the image of the multiprobe is roughly centered on the
+            # mppc detector.
+            logging.debug("Running image translation pre alignment before ROA acquisition.")
+            descanner_offset = image_translation_pre_align.run_image_translation_pre_align(self._scanner,
+                                                                                           self._multibeam,
+                                                                                           self._descanner,
+                                                                                           self._detector,
+                                                                                           self._detector.data,
+                                                                                           self._ccd)
+
+            # Set the descanner offset to the value calibrated with the image translation pre-alignment.
+            asm_config_orig["descanner"]["scanOffset"] = descanner_offset
+            logging.debug(f"Descanner offset set to: {descanner_offset}")
+            # Blank the beam to reduce beam damage on the sample.
+            self._scanner.blanker.value = True
+            # set back the initial values on the hardware
+            configure_hw.configure_asm(self._multibeam,
```

Comment:
Should this go into the finally? If the code fails beforehand, the system might be in a bad state right?

Reply:
that makes sense!